use-cases/rag-pipeline/data-preprocessing/README.md

```diff
@@ -0,0 +1,224 @@
+## Data Preprocessing for RAG
+
+### Dataset
+
+[This](https://www.kaggle.com/datasets/PromptCloudHQ/flipkart-products) is a pre-crawled public dataset, taken as a subset of a bigger dataset (more than 5.8 million products) that was created by extracting data from [Flipkart](https://www.flipkart.com/), a leading Indian eCommerce store.
+
+The dataset has product information such as id, name, brand, description, image urls, product specifications.
+
+In the following section, you will run a GKE job to perform data preprocessing for RAG. The GKE job will run a python module named `datapreprocessing.preprocessing_rag` that does the following:
+
+- Read the dataset as a csv file from Cloud Storage
+- Clean up the product description text
+- Extract image urls, validate and download the images into Google Cloud Storage
+- Cleanup & extract attributes as key-value pairs
+- Filters data based on categories and value counts, selecting relevant columns, and removing duplicates
+- Uploads the processed data as a csv file to Google Cloud Storage
+
+The data preprocessing step takes approximately 18-20 minutes.
+
+### Prerequisites
+
+- This guide was developed to be run on the [playground AI/ML platform](/platforms/gke-aiml/playground/README.md). If you are using a different environment the scripts and manifest will need to be modified for that environment.
+- The raw data that will be processed in this example, follow [these instructions](/use-cases/prerequisites/raw-data.md) to load the data into the bucket.
+
+### Preparation
+
+- Clone the repository and change directory to the guide directory
+
+  ```shell
+  git clone https://github.com/GoogleCloudPlatform/accelerated-platforms && \
+  cd accelerated-platforms/use-cases/rag-pipeline/data-preprocessing
+  ```
+
+- Ensure that your `MLP_ENVIRONMENT_FILE` is configured
+
+  ```shell
+  cat ${MLP_ENVIRONMENT_FILE} && \
+  source ${MLP_ENVIRONMENT_FILE}
+  ```
+
+  > You should see the various variables populated with the information specific to your environment.
+
+### Build the container image
+
+- Build container image using Cloud Build and push the image to Artifact Registry
+
+  ```shell
+  cp -r ${MLP_BASE_DIR}/modules/python/src/datapreprocessing src/
+  cd src
+  sed -i -e "s|^serviceAccount:.*|serviceAccount: projects/${MLP_PROJECT_ID}/serviceAccounts/${MLP_BUILD_GSA}|" cloudbuild.yaml
+  gcloud beta builds submit \
+  --config cloudbuild.yaml \
+  --gcs-source-staging-dir gs://${MLP_CLOUDBUILD_BUCKET}/source \
+  --project ${MLP_PROJECT_ID} \
+  --substitutions _DESTINATION=${MLP_RAG_DATA_PROCESSING_IMAGE}
+  cd ..
+  rm -rf src/datapreprocessing
```

Comment:
Why are you removing the modules?

Reply:
just to avoid having 3 copies of the modules lying around in three different directories. It is not required to remove the modules but I thought it would be cleaner.